### PR TITLE
Update functions.php

### PR DESCRIPTION
**Pull request by nicodmf**
actually, the upload function don't take any options from deploy.php, but documentation mentions option for rsync.
This commit correct that.

| Q             | A
| ------------- | ---
| Bug fix?      | Yes
| New feature?  | No
| BC breaks?    | No
| Deprecations? | No
| Fixed tickets | N/A

### CODE REVIEW

**Comment by antonmedv** [comment]
What is this for?

**Comment by Exotherme** [comment]
@antonmedv 
I just ran into this issue myseld and found this issue. The problem is that if you want to add your own options to rsync, in my case "--inplace", it will not be added because the 'options' config is always overwritten if there are SSH arguments.

File: src/functions.php [hunk 1]
```diff
@@ -423,13 +423,14 @@ function upload($source, $destination, array $config = [])
...
     $source = parse($source);
     $destination = parse($destination);
+    $config['options'] = is_array($config['options']) ? $config['options'] : [];
 
     if ($host instanceof Localhost) {
         $rsync->call($host->getHostname(), $source, $destination, $config);
     } else {
         $sshArguments = $host->getSshArguments()->getCliArguments();
         if (empty($sshArguments) === false) {
-            $config['options'] = ["-e 'ssh $sshArguments'"];
```
Comment (Exotherme) [inline src/functions.php]: Alternatively add the array check here:
if (!is_array($config['options'])) {
    $config['options'] = [];
}

Comment (nicodmf) [inline src/functions.php]: as it, i think options are too for localhost

**Comment by nicodmf** [comment]
actually, the upload function don't take any options from deploy.php, but documentation mentions option for rsync.
This commit correct that.

**Comment by Exotherme** [comment]
@antonmedv & @nicodmf 
Unfortunately this change now causes a PHP notice:
PHP Notice:  Undefined index: options in src/functions.php on line 426
PHP Notice:  Undefined index: options in src/functions.php on line 452

**Comment by antonmedv** [comment]
@nicodmf please, fix

**Comment by Exotherme** [comment]
@antonmedv I've created a fix: #1264

**Pull request merged**